resource.address/tcp/src/main/java/org/kaazing/gateway/resource/address/tcp/TcpResourceAddress.java

```diff
@@ -27,41 +27,45 @@
 import org.kaazing.gateway.resource.address.ResourceOption;
 
 public final class TcpResourceAddress extends ResourceAddress {
-	
-	private static final long serialVersionUID = 1L;
 
-	static final String TRANSPORT_NAME = "tcp";
-	
-	public static final ResourceOption<InetSocketAddress> BIND_ADDRESS = new TcpBindAddressOption();
-	public static final ResourceOption<Long> MAXIMUM_OUTBOUND_RATE = new TcpMaximumOutboundRateOption();
+    private static final long serialVersionUID = 1L;
 
-	private static final long MAXIMUM_OUTBOUND_RATE_DEFAULT = 0xFFFFFFFFL;
-	
-	private InetSocketAddress bindAddress;
-	private long maximumOutboundRate = MAXIMUM_OUTBOUND_RATE.defaultValue();
+    static final String TRANSPORT_NAME = "tcp";
 
-	TcpResourceAddress(ResourceAddressFactorySpi factory, String original, URI resource) {
-		super(factory, original, resource);
-	}
+    public static final ResourceOption<InetSocketAddress> BIND_ADDRESS = new TcpBindAddressOption();
+    public static final ResourceOption<Long> MAXIMUM_OUTBOUND_RATE = new TcpMaximumOutboundRateOption();
+    public static final ResourceOption<Long> HANDSHAKE_TIMEOUT = new HandshakeTimeoutOption();
 
-	@Override
-	@SuppressWarnings("unchecked")
+    private static final long MAXIMUM_OUTBOUND_RATE_DEFAULT = 0xFFFFFFFFL;
+
+    private InetSocketAddress bindAddress;
+    private long maximumOutboundRate = MAXIMUM_OUTBOUND_RATE.defaultValue();
+    private Long handshakeTimeout = HANDSHAKE_TIMEOUT.defaultValue();
+
+    TcpResourceAddress(ResourceAddressFactorySpi factory, String original, URI resource) {
+        super(factory, original, resource);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
     protected <V> V getOption0(ResourceOption<V> option) {
         if (option instanceof TcpResourceOption) {
-            TcpResourceOption tcpOption = (TcpResourceOption)option;
+            TcpResourceOption tcpOption = (TcpResourceOption) option;
             switch (tcpOption.kind) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaazing.wan:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)